resources/js/features/forums/components/RecentPostsMainRoot/RecentPostsPagination/RecentPostsPagination.tsx

```diff
@@ -0,0 +1,39 @@
+import { usePage } from '@inertiajs/react';
+import type { FC } from 'react';
+
+import {
+  BasePagination,
+  BasePaginationContent,
+  BasePaginationItem,
+  BasePaginationNext,
+  BasePaginationPrevious,
+} from '@/common/components/+vendor/BasePagination';
+import type { RecentPostsPageProps } from '@/features/forums/models';
+
+export const RecentPostsPagination: FC = () => {
+  const { previousPageUrl, nextPageUrl, maxPerPage } = usePage<RecentPostsPageProps>().props;
+
+  if (!previousPageUrl && !nextPageUrl) {
+    return null;
+  }
+
+  return (
+    <BasePagination>
+      <BasePaginationContent>
+        {previousPageUrl ? (
+          <BasePaginationItem>
+            <BasePaginationPrevious href={previousPageUrl}>
+              Previous {maxPerPage}
+            </BasePaginationPrevious>
+          </BasePaginationItem>
+        ) : null}
+
+        {nextPageUrl ? (
+          <BasePaginationItem>
+            <BasePaginationNext href={nextPageUrl}>Next {maxPerPage}</BasePaginationNext>
```

Comment:
Stylistically, shouldn't this be three lines to match up with the Previous implementation above?

Reply:
This formatting is left entirely at the mercy of ESLint & Prettier. This is beneficial, as we don't need to really focus on code formatting & style. Prettier can take the wheel.

When editing JS/TS, I always leave VSCode's "format on save" setting enabled:


https://github.com/user-attachments/assets/a90112e4-889f-4687-99ef-e435b8b43feb

Reply:
Another example of Prettier auto-formatting. It's pretty normal to code something as a stream of consciousness and then hit save and let Prettier do all the work. Note that these Prettier rules have been in place since https://github.com/RetroAchievements/RAWeb/pull/2258.


https://github.com/user-attachments/assets/ae64d77c-28ae-4086-b7ae-6de7d2b7bed9